src/Microsoft.Identity.Client/Internal/OAuth2/OAuth2Client.cs

```diff
@@ -88,18 +88,34 @@ public void AddBodyParameter(string key, string value)
             }
 
             HttpResponse response = null;
-
-            if (method == HttpMethod.Post)
+            Uri endpointUri = CreateFullEndpointUri(endPoint);
+            var httpEvent = new HttpEvent()
+            {
+                HttpPath = EventBase.ScrubTenant(endpointUri),
+                QueryParams = String.Join("&", MsalHelpers.ParseKeyValueList(endpointUri.Query, '&', false, true, requestContext).Keys)
```

Comment:
IMHO, "you should pass null as request context anyways", that would be a very surprising move which is not considered as general practice. In general, when a code snippet (a.k.a. the caller) calls another function, the caller needs to supply all the parameters required by that function's signature, otherwise such code snippet is probably not the best place to make that function call.

Here in this particular case, the `requestContext` parameter is defined in ALL of the `ParseKeyValueList()` overload's signature. So there is no clear sign that it would be an optional parameter, or it would welcome a `null` input. When I wrote the current implementation, I did not know/expect it would accept `null`.

Now that you brought this up, I went to take a look into its implementation detail and now know `null` is also valid. So technically we can put this `ParseKeyValueList(...)` thing inside the setter, if we really want to. But I would say, the fact that a caller developer need to spend extra time to read into the helper's implementation detail to gain prior-knowledge, is not a normal workflow.